src/graph/GraphViewServer.ts

```diff
@@ -332,14 +362,26 @@ export class GraphViewServer extends EventEmitter {
       originalHandleProtocolMessage.call(this, message);
     };
 
+    let streamError = undefined;
+    client.on('error', error => handleError(error));
+
+    function handleError(err) {
+      console.warn(err);
+      streamError = err;
+    }
+
     return new Promise<[{}[]]>((resolve, reject) => {
       client.execute(gremlinQuery, {}, (err, results) => {
-        if (err) {
+        if (streamError) {
+          this.log("Gremlin client error: ", streamError.message || streamError.toString());
+          reject(streamError);
+        } else if (err) {
```

Comment:
Not quite sure what the difference between `streamError` and `err` is. Could you give an example of each?

Reply:
Essentially streamError is anything that the gremlin client is handling itself, i.e. errors from websocket/etc.  I'll change the name to that.  Personally I think they should be routing everything the same way.